### PR TITLE
code quality

### PR DESCRIPTION
**Pull request by ESadek-MO**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,60 +1,192 @@
...
 
+#[tool.pytest.ini_options]
+#minversion = "6.0"
+#addopts = ["-ra", "--showlocals", "--strict-markers", "--strict-config"]
+#xfail_strict = true
+#filterwarnings = ["error"]
+#log_cli_level = "info"
+#testpaths = [
+#  "tephi",
+#]
```
Comment (stephenworsley) [inline pyproject.toml:64]: I think it would be better to leave this section out entirely as it hasn't been tested yet.

File: pyproject.toml [hunk 2]
```diff
@@ -1,60 +1,192 @@
...
+    # -> No extended summary found
+    "ES01",  # Not all docstrings require an "Extended Summary" section.
+
+    # -> No examples section found
+    "EX01",  # Not all docstrings require an "Examples" section.
+
+    # -> No Yields section found
+    "YD01",  # Not all docstrings require a "Yields" section.
+
+    # TODO: exceptions that still need investigating are below. Might be fixable, or might become permanent (above):
```
Comment (stephenworsley) [inline pyproject.toml:133]: This line can probably be removed now.

File: pyproject.toml [hunk 3]
```diff
@@ -1,60 +1,192 @@
...
+
+    # flake8-implicit-str-concat (ISC)
+    # https://docs.astral.sh/ruff/rules/single-line-implicit-string-concatenation/
+    # NOTE: This rule may cause conflicts when used with "ruff format".
+    "ISC001",  # Implicitly concatenate string literals on one line.
+
+
+    # TODO: exceptions that still need investigating are below. Might be fixable, or might become permanent (above):
+    "ERA001", # Has commented out code
+    "EXE001", # Shebang is present but file is not executable
```
Comment (stephenworsley) [inline pyproject.toml:163]: If this is limited to setup.py (as it was in iris-esmf-regrid), it may be worth adding to a per-file-ignores.

**Review by stephenworsley — CHANGES_REQUESTED** [review]
Looking good, couple comments here

File: pyproject.toml (hunk 3 above)
Comment (ESadek-MO) [reply pyproject.toml]: yep, that's a good call. I've also added ERA001 to conf.py, as the commenting out is intentional there (for documentary purposes)

File: pyproject.toml (hunk 2 above)
Comment (ESadek-MO) [reply pyproject.toml]: https://github.com/SciTools/tephi/pull/172/commits/c0eeb8066621debbf4aa3444c3ddc7ce153179e9

File: pyproject.toml (hunk 1 above)
Comment (ESadek-MO) [reply pyproject.toml]: https://github.com/SciTools/tephi/pull/172/commits/c0eeb8066621debbf4aa3444c3ddc7ce153179e9

File: pyproject.toml [hunk 4]
```diff
@@ -1,60 +1,189 @@
...
+# for SciTools linting guidelines
+
 [tool.black]
 line-length = 79
 target-version = ["py37", "py38", "py39", "py310"]
 include = '\.pyi?$'
 
+[project]
+authors = [
+    {name = "UK Met Office", email = "scitools-iris-dev@googlegroups.com"}
```
[the remark this answers is not in the document]
Comment (ESadek-MO) [reply pyproject.toml]: https://github.com/SciTools/tephi/pull/172/commits/c0eeb8066621debbf4aa3444c3ddc7ce153179e9

File: pyproject.toml (hunk 3 above)
Comment (ESadek-MO) [reply pyproject.toml]: https://github.com/SciTools/tephi/pull/172/commits/dd417efbb5d163771e07dde9694030e11f0c8d0e

**Review by stephenworsley — APPROVED** [review]
Looks good.

**Pull request merged**